test/arcilator/serialize-state-info.mlir

```diff
@@ -1,5 +1,4 @@
-// RUN: circt-opt %s --arc-print-state-info=state-file=%t
-// RUN: cat %t | FileCheck %s
+// RUN: (arcilator %s --state-file=%t --until-before=preproc > /dev/null) && (cat %t | FileCheck %s)
```

Comment:
That's quite hacky. It would be a lot nicer if the analysis+printing could be invoked as an individual pass, e.g., by adding the registration code to `circt-translate`. I'd not be too worried that it is not a full 'translation' but just translates the state layout to a JSON, because `circt-translate` and `circt-opt` are just test driver binaries anyways. Being able to invoke it separately from the CLI could also be useful for debugging and experimenting.

Reply:
Okay, I added a translation lib for it. Although now it makes me realize the `--state-file` flag is a bit gross, especially considering that `--split-input-file` is accepted by arcilator. However this is already the case anyway, if changes need to be made they can be done in a further PR.